### DISCUSSION TITLE
Task status is up_for_retry but log shows Skipped and downstream task is running.

### DISCUSSION THREAD

**Discussion by baidoosik** [apache/airflow#28674] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Airflow version: 2.4.1

Upstream task status is `up_for_retry` but in the task log shows marked `skipped`.
and downstream task was skipped.(means executed) although upstream task status is `up_for_retry`.
trigger rule is default value. (all_success).

There are 3 issues.

1. log said skipped. but task status is up_for_retry.
2. although status is up_for_retry, task never don't reschedule.
3. downstream task is running although upstream task status is up_for_retry.

![airflow_issue1](https://user-images.githubusercontent.com/23095231/207788380-4dc54324-1b78-44b4-a788-b1cb028d8fce.png)

![airflow_issue2](https://user-images.githubusercontent.com/23095231/207788412-1f3d511c-d3f5-4065-bb45-aa59399e4319.png)

code makes SkipException.
```
....
    if resp["KeyCount"] == 0:
        logging.info("No incremental file detected. skip the task.")
        raise AirflowSkipException
```

### What you think should happen instead

1. log said skipped. but task status is up_for_retry. -> task was skipped or log says up_for_retry.
2. although status is up_for_retry, task never don't reschedule. -> reschedule and re-run.
3. downstream task is running although upstream task status is up_for_retry. -> downstream task can't run.

### How to reproduce

I don't know the way to reproduce it. sorry.

### Operating System

docker image: apache/airflow:2.4.1-python3.9

### Versions of Apache Airflow Providers

apache-airflow[pre-commit]==2.4.1
apache-airflow[pandas]==2.4.1
apache-airflow[google]==2.4.1
apache-airflow[apache-airflow-providers-google]==2.4.1
apache-airflow[dask]==2.4.1
apache-airflow[google-cloud-bigquery]==2.4.1
apache-airflow[sentry]==2.4.1
apache-airflow[slack]==2.4.1
fastparquet==0.8.1
s3fs==2022.7.1

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!
